Source/Public/Add-VSTeamTaskGroup.ps1

```diff
@@ -0,0 +1,28 @@
+function Add-VSTeamTaskGroup {
+   [CmdletBinding()]
+   param(
+      [Parameter(ParameterSetName = 'ByFile', Mandatory = $true, ValueFromPipelineByPropertyName = $true)]
+      [string] $InFile,
+
+      [Parameter(ParameterSetName = 'ByBody', Mandatory = $true, ValueFromPipelineByPropertyName = $true)]
+      [string] $Body
+   )
+
+   DynamicParam {
+      _buildProjectNameDynamicParam
+   }
+
+   process {
+      # Bind the parameter to a friendly variable
+      $ProjectName = $PSBoundParameters["ProjectName"]
+
+      if ($InFile) {
+         $resp = _callAPI -Method Post -ProjectName $ProjectName -Area distributedtask -Resource taskgroups -Version $([VSTeamVersions]::TaskGroups) -InFile $InFile -ContentType 'application/json'
+      }
+      else {
+         $resp = _callAPI -Method Post -ProjectName $ProjectName -Area distributedtask -Resource taskgroups -Version $([VSTeamVersions]::TaskGroups) -ContentType 'application/json' -Body $Body
+      }
+
+      return $resp
```

Comment:
Are you missing a call to _applyTypesToVariableGroup?

Reply:
I actually omitted that on purpose.  I noticed at least one Get function in which you're not doing _applyTypesTo{Type}; I forget which function I was looking at.  Because it seemed a bit "extra," and isn't necessary for my use case, I honestly never sat down and figured out what _applyTypesTo{Type} is actually doing and didn't implement _applyTypesToTaskGroup.  In any case, are you requesting that I implement and use _applyTypesToTaskGroup?

Reply:
What the call does is assigns a type name to the object so Formats can be applied.  For example if you look into the formats folder there ps1xml files in there. They decide which are the default properties to display when an object is written to the console. You get to control what the table form, list for and others contain. If you don't assign a type the object is just a generic object and a format can't be located for that.

Reply:
I will take care of it.

Reply:
Oh, OK, thanks!